File: frontend/types/sentiment.ts
// Mirrors backend/app/schemas/sentiment.py

export type SentimentLabel = "positive" | "negative" | "neutral" | "mixed";

export interface SentimentItem {
  title: string;
  publisher: string | null;
  url: string | null;
  published_at: string | null;
  label: "positive" | "negative" | "neutral";
  score: number;
}

export interface Sentiment {
  ticker: string;
  overall_label: SentimentLabel;
  overall_score: number;
  headline_count: number;
  counts: { positive: number; negative: number; neutral: number };
  summary: string;
  items: SentimentItem[];
  note: string;
}

File: backend/app/schemas/sentiment.py
from pydantic import BaseModel


class SentimentItem(BaseModel):
    title: str
    publisher: str | None = None
    url: str | None = None
    published_at: str | None = None
    label: str  # positive / negative / neutral
    score: float


class SentimentCounts(BaseModel):
    positive: int
    negative: int
    neutral: int


class SentimentOut(BaseModel):
    ticker: str
    overall_label: str  # positive / negative / neutral / mixed
    overall_score: float  # -1..1
    headline_count: int
    counts: SentimentCounts
    summary: str
    items: list[SentimentItem]
    note: str
